cGame.h

```diff
@@ -103,10 +103,8 @@ class cGame : public cScenarioObserver, cInputObserver {
     // Use this instead
 	void setNextStateToTransitionTo(int newState);
 
-	int getMaxVolume() { return iMaxVolume; }
-
-	cSoundPlayer * getSoundPlayer() {
-	    return soundPlayer;
+	cSoundPlayer& getSoundPlayer() {
+	    return *_soundplayer;
```

Comment:
so this is something I haven't decided yet for naming. I see several approaches, and the google c++ guides also has an opinion about it (but I don't like it).

So these are the possibilities:
`m_<var name>`
`<var name>_`
`_<var name>`

I believe google opts for the second version (appending an underscore).

Personally I don't like either of these options, but you get warnings/errors when you shadow variable names. And being explicit with `this->x = x` is cumbersome perhaps (although fairly common in java).

So what shall we choose?

I lean towards `m_<somevar>` now.

Reply:
> Also interesting, so this function returns the hard pointer, and disregarding the fact it is a `unique_ptr`?

We can tell the caller that they don't have to do a nullptr check. I envision cSoundPlayer to be constructed in the constructor of cGame, then we are certain it's always initialized when we need it. However, cGame is still a global variable, so that is not possible yet...

Reply:
> I lean towards `m_<somevar>` now.

I've used both `m_<something>` and `<something>_` professionally. The m_ is a bit better to see.

Reply:
ok, I suggest to use `m_something` then.

Reply:
> > Also interesting, so this function returns the hard pointer, and disregarding the fact it is a `unique_ptr`?
> 
> We can tell the caller that they don't have to do a nullptr check. I envision cSoundPlayer to be constructed in the constructor of cGame, then we are certain it's always initialized when we need it. However, cGame is still a global variable, so that is not possible yet...

I agree 👍